### PR TITLE
add async/await, es6 structure and multiple improvements

### PR DESCRIPTION
**Pull request by nacimgoura**
I've made many changes:
- Remove bluebird dependencies
- Replace request by axios
- Reformat code with es6 syntax
- Use async/await
- Clearer code
- Multiple improvements
- Update README

There is still code that can be refactorized but I did most of it.

Due to many changes I could not test everything. I did `npm test`, and test the overall functioning of the application and everything seems to work. 

Relative issues : #720, #698

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,6 @@
     "multer": "^1.3.0",
     "node-foursquare": "^0.3.2",
     "node-linkedin": "^0.5.6",
-    "node-sass": "^4.7.2",
```
Comment (YasharF) [inline package.json]: Why is nore-sass being removed?

**Comment by nacimgoura** [comment]
I thought I had detailed it. Because it's already declared in peerDependencies.

**Comment by nacimgoura** [comment]
Thank you for your feedback, but I don't understand some things :
- The only modification for controllers/home.js is the es6 structure with implicit return.
- User model doesn't changes (there is just a line break for clarity but no impact on operation).
- I remove node-sass because it's declared twice in package.json.
- 
```
Console.log messaging when the server starts, out of scope. There are issues with that line as it currently is, but there is more work that needs to be done to fix that line.
```
I don't understand this sentence, can you give me more details?


I am not submitting untested changes. I run an `npm test` and test myself before publishing. But there are some APIs (like instagram) that require external validation and so I can't test them.

From now on I will make small pull requests.

**Comment by YasharF** [comment]
> The only modification for controllers/home.js is the es6 structure with implicit return.
> User model doesn't changes (there is just a line break for clarity but no impact on operation).
> console.log messaging when the server starts

From the standpoint of a reviewer or a post-hackathon user (doing rebase or picking diffs), why was the original line _broken_ in the first place requiring a change?  This is specially important from a post-hackathon user considering that there might be merge conflicts that they need to figure out.  Is it addressing a bug?  Is it something that they need to look at the rest of their code and replicate throughout?  Is it just code-style change?  If it is code-style, why was it not important enough to be flagged by eslint?  Can that change by itself justify its need in a standalone PR?  For example, is there currently no implicit return in the specific place, and what is the problem that it is causing?
When having smaller commits and PRs that are focused on specific problems instead of a bundled case, these can be clarified one by one.  Take a look at the commit description for https://github.com/sahat/hackathon-starter/commit/e9f0e8e673cbf9b0f8dd0aef808da14403ab3b4b ; even though there are only 7 lines of code that have changed, there is a much longer commit description (158 words) that has been added to justify why the change is needed and without it the code base has bugs.

> I remove node-sass because it's declared twice in package.json

What problem was this causing?  Was there a warning in npm?  Since this involves the build process, it requires verification that it doesn't break builds on supported stacks: Heroku, OpenShift, Azure, Bluemix, GCP

> I am not submitting untested changes. I run an npm test and test myself before publishing. But there are some APIs (like instagram) that require external validation and so I can't test them.

Currently there is no coverage for most issues with the current test cases, and `npm test` is not sufficient for testing.  Most changes do need to be manually tested.  In case of the APIs, they need to be tested using real keys and real accounts for various cases that the API example may run into.  In other words, you do need to run the server and make sure that no functionality.  Another benefit of having smaller focused PRs is that, since you need to manually test your changes, you can focus on a smaller number of manual checks that cover the scope of your changes before submitting the PR.

**Comment by nacimgoura** [comment]
Okay, I understand. Do not merge or close this pull request, I leave it there to serve as reference. After this, we can close it.

**Comment by YasharF** [comment]
Is there anything left on this to do or can we close this PR?

**Comment by nacimgoura** [comment]
Yes, there's still :
- use of promises instead of callbacks for mongo requests
- possibly the use of axios instead of request

**Comment by YasharF** [comment]
Lets reduce the scope to just 

> use of promises instead of callbacks for mongo requests

so we can close this, and we'll revisit RequestJS vs Axios at a later point.

**Pull request closed without merging**